tasks/fix-cat2.yml

```diff
@@ -325,14 +327,51 @@
       - RHEL-07-010260
       - password
 
-- name: "MEDIUM | RHEL-07-010270 | PATCH | Passwords must be prohibited from reuse for a minimum of five generations."
-  pamd:
-      name: system-auth
-      state: args_present
-      type: password
-      control: sufficient
-      module_path: pam_unix.so
-      module_arguments: "remember=5"
+- name: "MEDIUM | RHEL-07-010270 | PATCH | The Red Hat Enterprise Linux operating system must be configured so that passwords are prohibited from reuse for a minimum of five generations."
+  block:
+      - name: "MEDIUM | RHEL-07-010270 | PATCH | Ensure pam_pwhistory rule exists"
+        pamd:
+            name: "{{ item }}"
+            state: before
+            type: password
+            control: sufficient
+            module_path: pam_unix.so
+            new_type: password
+            new_control: requisite
+            new_module_path: pam_pwhistory.so
+            module_arguments:
+              - use_authtok
+              - remember={{ rhel7stig_pam_pwhistory.remember|default(5) }}
+              - retry={{ rhel7stig_pam_pwhistory.retries|default(3) }}
+        with_items:
+            - "system-auth"
+            - "password-auth"
+
+      # Temporary audit check to determine if the following pamd module task needs to be run since the module is not idempotent
+      - name: "MEDIUM | RHEL-07-010270 | AUDIT | Check for existing password history reuse settings"
```

Comment:
I'd drop this task entirely.

Reply:
That's a bug in the pamd module, then...

Reply:
So I think I need to keep this task until the pamd module can be fixed and then I have 1 of 2 options. Keep using the `updated` task below to make the change (this works correctly with argument value changes) OR I can move to `args_present` but if I do that I'll need to add ANOTHER task to remove all the args before I re-add them.

The current way seems like the least hacky way until we can get the pamd module fixed.